bee-ternary/src/bigint/overflowing_add.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+//! Extensions to `overflowing_add`.
+
+pub(crate) trait OverflowingAdd<T = Self> {
+    /// Extends `overflowing_add` with a carry.
+    fn overflowing_add_with_carry(self, other: T, carry: T) -> (T, bool);
+}
+
+impl OverflowingAdd for u32 {
+    fn overflowing_add_with_carry(self, other: Self, carry: Self) -> (Self, bool) {
+        let (sum, first_overflow) = self.overflowing_add(other);
```

Comment:
I'm not entirely sure what the purpose of this code is, could you explain? I see that there's an add-plus-carry but I'm sure what context this all might be useful in.

Reply:
I am not the author of the code but I'll try to explain.
When dealing with big integers you have a lot of carry propagation to handle. This piece of code is just an extension of the overflowing add with a carry. It could technically be done without this function but this is just to DRY things I assume.